src/nft/components/details/TraitsContainer.tsx

```diff
@@ -0,0 +1,99 @@
+import { GenieAsset, Trait } from 'nft/types'
+import qs from 'query-string'
+import { useMemo } from 'react'
+import { Link } from 'react-router-dom'
+import styled from 'styled-components/macro'
+
+const Grid = styled.div`
+  display: grid;
+  grid-template-columns: 1fr 1fr 1fr 1fr;
+  gap: 16px;
+  max-width: 780px;
+
+  @media (max-width: 960px) {
+    grid-template-columns: 1fr 1fr 1fr;
+  }
+
+  @media (max-width: 420px) {
+    grid-template-columns: 1fr 1fr;
+  }
+`
+
+const GridItemContainer = styled(Link)`
+  background-color: ${({ theme }) => theme.backgroundInteractive};
+  border-radius: 12px;
+  cursor: pointer;
+  padding: 12px;
+  text-decoration: none;
+
+  &:hover {
+    opacity: ${({ theme }) => theme.opacity.hover};
+  }
+
+  &:active {
+    opacity: ${({ theme }) => theme.opacity.click};
+  }
+
+  transition: ${({
+    theme: {
+      transition: { duration, timing },
+    },
+  }) => `opacity ${duration.medium} ${timing.ease}`};
+  min-width: 0;
+`
+
+const TraitType = styled.div`
+  color: ${({ theme }) => theme.textSecondary};
+  font-weight: 600;
+  font-size: 10px;
+  line-height: 12px;
+  white-space: nowrap;
+  width: 100%;
+`
+
+const TraitValue = styled.div`
+  color: ${({ theme }) => theme.textPrimary};
+  font-size: 16px;
+  line-height: 24px;
+  margin-top: 4px;
+  display: inline-block;
+
+  display: inline-block;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+  max-width: 100%;
+`
+
+const GridItem = ({ trait, collectionAddress }: { trait: Trait; collectionAddress: string }) => {
+  const { trait_type } = trait
+  // @ts-ignore
+  const trait_value = trait.trait_value || trait.value
```

Comment:
if you implement my fix in src/graphql/data/nftDetails about renaming this param trait_value lns69-71 can be replaced with: `const { trait_type, trait_value } = trait`

Reply:
this is to accommodate the v3 endpoint since it's at the ```value``` key

Reply:
If you modify the v3 fetcher to cast the value to trait_value everything can use the Trait obj